### ISSUE TITLE
The `jsdoc/no-restricted-syntax` rule doesn't work with a single line JSDoc comment

### ISSUE THREAD

**Issue by macku** [gajus/eslint-plugin-jsdoc#831]
## Expected behavior
The `jsdoc/no-restricted-syntax` rule works with an inline and a single line JSDoc blocks.

## Actual behavior
The `jsdoc/jsdoc/no-restricted-syntax` doesn't report the issue when using an inline or single line JSDoc comment block. The rule works only for a multiline JSDdoc comment:

<img width="1318" alt="Screen Shot 2022-01-17 at 09 36 18" src="https://user-images.githubusercontent.com/227305/149683423-225fc8b6-8041-4065-89ce-0b4ea4fbb330.png">
<img width="1325" alt="Screen Shot 2022-01-17 at 09 35 46" src="https://user-images.githubusercontent.com/227305/149683424-507d8517-4c72-4c3e-9875-8a043f89af56.png">

## ESLint Config

```js
module.exports = {
  rules: {
    'jsdoc/no-restricted-syntax': [
      'error',
      {
        contexts: [
          {
            comment: 'JsdocBlock:has(JsdocTag[tag="type"]:has([value=/FunctionComponent/]))',
            context: 'any',
            message: 'The `FunctionComponent` type is not allowed. Please use `FC` instead.',
          },
        ],
      },
    ],
  }
};
```
## ESLint sample

```js
/** @type {React.FunctionComponent<{ children: React.ReactNode }>}*/
const MyComponent = ({ children }) => {
   return children;
}
```

## Environment

- Node version: v16.13.0
- ESLint version v8.7.0
- `eslint-plugin-jsdoc` version: 37.6.1

**Comment by macku** [comment]
I was trying to fix that over the weekend with the `jsdoccomment` project: 
https://github.com/es-joy/jsdoccomment/pull/6
